src/npu/burst_engine.py

```diff
@@ -352,9 +352,9 @@ def ipu_handler(ipu_data):
             for sensor_type in ipu_data:
                 # Ultrasonic / Lidar Handler
                 # todo: need a more consistent naming convention when it comes to lidar vs ultrasonic vs proximity
-                if sensor_type == 'ultrasonic':
+                if sensor_type == 'ultrasonic0':
```

Comment:
to make this generic you can have it as follows
`if sensor_type like 'ultrasonic`
this way you will catch all the extended enumerations